### PR TITLE
feat: Add Timeline.Item.position (#16148)

### LINKED ISSUE #16177: Timeline.pendingDot interface documentation there is a small problem
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/timeline/#Timeline](https://ant.design/components/timeline/#Timeline)

### Steps to reproduce
check the type of pendingDot

### What is expected?
string|ReactNode

### What is actually happening?
|string|ReactNode

| Environment | Info |
|---|---|
| antd | 3.16.3 |
| React | None |
| System | None |
| Browser | None |

### PR DESCRIPTION
**Pull request by MrHeer**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

close #16177 and #16148 

### 💡 Solution

reuse 'alternate' less

### 📝 Changelog

add Timeline.Item.position

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/timeline/index.en-US.md](https://github.com/MrHeer/ant-design/blob/feature-issues#16148/components/timeline/index.en-US.md)
[View rendered components/timeline/index.zh-CN.md](https://github.com/MrHeer/ant-design/blob/feature-issues#16148/components/timeline/index.zh-CN.md)

### CODE REVIEW

File: components/timeline/Timeline.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export interface TimelineProps {
   pendingDot?: React.ReactNode;
   style?: React.CSSProperties;
   reverse?: boolean;
-  mode?: 'left' | 'alternate' | 'right';
+  mode?: 'left' | 'alternate' | 'right' | 'customize';
```
Comment (afc163) [inline components/timeline/Timeline.tsx]: No need to add `mode="customize"`, just follow postion of item in higher priority.

Comment (MrHeer) [reply components/timeline/Timeline.tsx]: That goot idea

Comment (MrHeer) [reply components/timeline/Timeline.tsx]: I think the mode shuld be in higher priority.

**Comment by afc163** [comment]
You should pull and rebase `feature` branch to fix ci.

**Comment by MrHeer** [comment]
> You should pull and rebase `feature` branch to fix ci.

I did rebase, I'll try again tonight. It is necessary to rewrite history(Squash `bd925d4` and `e28570b`)?

**Comment by afc163** [comment]
Not necessary but better~

**Comment by MrHeer** [comment]
Thank you very much.

File: components/timeline/TimelineItem.tsx [hunk 2]
```diff
@@ -58,6 +60,7 @@ const TimelineItem: React.SFC<TimeLineItemProps> = props => (
 TimelineItem.defaultProps = {
   color: 'blue',
   pending: false,
+  position: 'left',
```
Comment (afc163) [inline components/timeline/TimelineItem.tsx]: should follow `<Timeline mode="xxx">`, don't set in defaultProps.

Comment (MrHeer) [reply components/timeline/TimelineItem.tsx]: After the mode is set, the position will be invalid.

File: components/timeline/Timeline.tsx [hunk 3]
```diff
@@ -74,9 +74,13 @@ export default class Timeline extends React.Component<TimelineProps, any> {
...
               ? `${prefixCls}-item-left`
               : `${prefixCls}-item-right`
+            : mode === 'left'
+            ? `${prefixCls}-item-left`
             : mode === 'right'
             ? `${prefixCls}-item-right`
-            : '',
+            : ele.props.position === 'right'
+            ? `${prefixCls}-item-right`
+            : `${prefixCls}-item-left`,
```
Comment (zombieJ) [inline components/timeline/Timeline.tsx]: 这段代码原本就挺看不懂，现在更难懂了。抽出来吧。

Comment (MrHeer) [reply components/timeline/Timeline.tsx]: I think so

**Comment by MrHeer** [comment]
@zombieJ I have modified. Is there other problem?

**Comment by zombieJ** [comment]
https://deploy-preview-16193--ant-design.netlify.com/components/timeline-cn/#header
Seems style different with current:
http://ant.design/components/timeline-cn/

**Comment by MrHeer** [comment]
Yes, I have to change the style to accommodate that you can set the position of the dot in the case of default.

**Comment by zombieJ** [comment]
We should not break origin behavior.

**Comment by MrHeer** [comment]
Yeah, I understand, so I started to consider adding a mode `customize`  (#16148). but afc163 didn't think it is necessary (just follow postion of item in higher priority.)

I think you have to add a new mode except you have a good plan

**Comment by afc163** [comment]
We can use `mode="alternate"` as the custom mode.

**Comment by MrHeer** [comment]
> We can use `mode="alternate"` as the custom mode.

Haha, nice, I also consider it, but feel a bit strange, so I add a `customize`(as same as `alternate`) mode

**Comment by MrHeer** [comment]
I've updated, if you have time, please review it. Thank you!

File: components/timeline/Timeline.tsx [hunk 4]
```diff
@@ -55,6 +55,18 @@ export default class Timeline extends React.Component<TimelineProps, any> {
       ? [pendingItem, ...React.Children.toArray(children).reverse()]
       : [...React.Children.toArray(children), pendingItem];
 
+    const getPositionCls = (ele: React.ReactElement<any>, idx: number) => {
```
Comment (zombieJ) [inline components/timeline/Timeline.tsx]: `ele` => `{ props: { position } }`, `idx` => `index`

File: components/timeline/index.en-US.md [hunk 5]
```diff
@@ -41,3 +41,4 @@ Node of timeline
 | -------- | ----------- | ---- | ------- |
 | color | Set the circle's color to `blue`, `red`, `green` or other custom colors | string | `blue` |
 | dot | Customize timeline dot | string\|ReactNode | - |
+| position | Customize node position | `left` \| `right` | - |
```
Comment (zombieJ) [inline components/timeline/index.en-US.md]: One more column about avaliable `version`:
屏幕快照 2019-04-26 上午10 56 36

Comment (MrHeer) [reply components/timeline/index.en-US.md]: OK, I'll update tonight

Comment (MrHeer) [reply components/timeline/index.en-US.md]: `version` should be `3.16.6`?

Comment (zombieJ) [reply components/timeline/index.en-US.md]: feature branch always released with minor version. Use `3.17.0` instead.

Comment (MrHeer) [reply components/timeline/index.en-US.md]: Thank you

**Review by afc163 — APPROVED** [review]

**Pull request merged**